generator/src/main/java/com/scottlogic/deg/generator/ViolationGenerationEngineWrapper.java

```diff
@@ -0,0 +1,98 @@
+package com.scottlogic.deg.generator;
+
+import com.google.inject.Inject;
+import com.google.inject.name.Named;
+import com.scottlogic.deg.generator.constraints.Constraint;
+import com.scottlogic.deg.generator.constraints.grammatical.AndConstraint;
+import com.scottlogic.deg.generator.constraints.grammatical.ViolateConstraint;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.outputs.manifest.ManifestWriter;
+import com.scottlogic.deg.generator.outputs.targets.FileOutputTarget;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.text.DecimalFormat;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ViolationGenerationEngineWrapper {
+    private final GenerationEngine generationEngine;
+    private final FileOutputTarget fileOutputTarget;
+    private final Path outputPath;
+
+    @Inject
+    public ViolationGenerationEngineWrapper(@Named("outputPath") Path outputPath, GenerationEngine generationEngine, FileOutputTarget fileOutputTarget){
```

Comment:
Should this be a `FileOutputTarget` or the `OutputTarget` type.
If the former then should you remove the interface? There is only one implementation now.

Reply:
This is a good question.

For now the implementation relies on the FileOutputTarget specifically 
(like how it previously relied on the DirectoryOutputTarget). 
Because it has to new up files for each rule.

In the future this will need to change for different types of outputTargets.

Which will require a refactoring in the future.

I believe it wouldn't be wise to try and preempt what the future one will look like until we know the requirements in the future.

So yeah, for now i think you're right and we should remove the interface, This will require us to replace the test output target class with a Mock